core/src/main/java/org/elasticsearch/index/query/WrapperQueryBuilder.java

```diff
@@ -162,8 +162,8 @@ protected boolean doEquals(WrapperQueryBuilder other) {
     @Override
     protected QueryBuilder<?> doRewrite(QueryRewriteContext context) throws IOException {
         try (XContentParser qSourceParser = XContentFactory.xContent(source).createParser(source)) {
-            QueryParseContext parseContext = context.newParseContext();
-            parseContext.reset(qSourceParser);
+            qSourceParser.setParseFieldMatcher(context.indexSettings.getParseFieldMatcher());
```

Comment:
do we want to expose getParseFieldMatcher in QueryRewriteContext that takes it from indexSettings? then we would not need it in QueryShardContext as it's a subclass?